### PR TITLE
Test against errors on empty input from stdin

### LINKED ISSUE #940: Cwltool crashes when reading empty json input file from stdin
The code in main.py doesn't take into account this rare edge-case.

The crash in particular is because value of an object is not compared correctly against None:

```
echo '{}' | cwltool --debug https://www.ebi.ac.uk/~eduardo/my_workflow_1539097640017.cwl -
/home/psafont/.local/etc/virtualenvs/cwltool/bin/cwltool 1.0.20180923172926
I'm sorry, I couldn't load this CWL file.
The error was: 
Traceback (most recent call last):
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/cwltool/main.py", line 567, in main
    loadingContext.overrides_list, tool_file_uri)
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/cwltool/main.py", line 245, in load_job_order
    elif job_order_file:
UnboundLocalError: local variable 'job_order_file' referenced before assignment
```

When those comparisons get fixed:

```
$ echo '{}' | cwltool --non-strict --debug https://www.ebi.ac.uk/~eduardo/my_workflow_1539097640017.cwl -
/home/psafont/.local/etc/virtualenvs/cwltool/bin/cwltool 1.0.20180923172926

[...]

[Errno 2] No such file or directory: '/home/psafont/dev/ops/configuration/tools/workflow-service/-'
Traceback (most recent call last):
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/ref_resolver.py", line 149, in fetch_text
    with open(urllib.request.url2pathname(str(path)), encoding='utf-8') as fp:
FileNotFoundError: [Errno 2] No such file or directory: '/home/psafont/dev/ops/configuration/tools/workflow-service/-'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/ref_resolver.py", line 575, in resolve_ref
    doc = self.fetch(doc_url, inject_ids=(not mixin))
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/ref_resolver.py", line 927, in fetch
    text = self.fetch_text(url)
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/ref_resolver.py", line 154, in fetch_text
    raise RuntimeError(Text(err))
RuntimeError: [Errno 2] No such file or directory: '/home/psafont/dev/ops/configuration/tools/workflow-service/-'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/cwltool/main.py", line 296, in init_job_order
    MutableMapping, loader.resolve_ref(cmd_line["job_order"])[0])
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/ref_resolver.py", line 575, in resolve_ref
    doc = self.fetch(doc_url, inject_ids=(not mixin))
  File "/home/psafont/.local/etc/virtualenvs/cwltool/lib/python3.6/site-packages/schema_salad/sourceline.py", line 168, in __exit__
    raise self.makeError(six.text_type(exc_value))
RuntimeError: [Errno 2] No such file or directory: '/home/psafont/dev/ops/configuration/tools/workflow-service/-'
```

Note that this invocation should be functionally equivalent to `cwltool --non-strict https://www.ebi.ac.uk/~eduardo/my_workflow_1539097640017.cwl`, which actually works

### PR DESCRIPTION
**Pull request by psafont**
fixes #940

### CODE REVIEW

File: tests/test_empty_input.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import os
+from cStringIO import StringIO
+from tempfile import NamedTemporaryFile
+
+from cwltool.main import main
+from tests.util import needs_docker
+
+@needs_docker
+def test_empty_input():
+    workflow = '''
```
Comment (mr-c) [inline tests/test_empty_input.py]: Why not use `util.get_data` and keep it external?

Comment (psafont) [reply tests/test_empty_input.py]: I followed the style in tool_argparse, I can make them all external

**Comment by psafont** [comment]
That was unexpected :)

File: tests/test_empty_input.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+from cwltool.main import main
+from .util import get_data, needs_docker
+
+@needs_docker
```
Comment (mr-c) [inline tests/test_empty_input.py]: Really? Maybe just

```suggestion
@windows_needs_docker
```

File: tests/test_empty_input.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+from cwltool.main import main
+from .util import get_data, needs_docker
```
Comment (mr-c) [inline tests/test_empty_input.py]:
```suggestion
from .util import get_data, windows_needs_docker
```

**Comment by mr-c** [comment]
Thank you @psafont !
